app/resources/api/rest/customer/v1/customer_portal_access_profile_resource.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Api
+  module Rest
+    module Customer
+      module V1
+        class CustomerPortalAccessProfileResource < JSONAPI::Resource
```

Comment:
I think it should be inherited from Api::Rest::Customer::V1::BaseResource to apply athentication and use usual 
**self.apply_allowed_accounts** method to return profile allowed for customer